src/modules/users/users.service.ts

```diff
@@ -392,7 +388,7 @@ export class UsersService {
       }
 
       user.role = role;
-      user.onboardingSteps = onboardingSteps;
+      user.onboardingSteps = '1';
       await this.userRepository.save(user);
 
       return user;
```

Comment:
After updating the user's role, please generate new tokens for the user. This is necessary because the Roles decorator relies on information about the user's role from the access token.

Reply:
done